File: WordExplorer/src/WordListInterface.java
/*	ICS499
 *  Group 2
 *  Main.java
 *  This is the Main program the starts program and builds GUI
 *  Please make sure jl1.0.1.jar is included in project build path
 */

import java.util.List;

public interface WordListInterface {

	// Interface Methods
	public boolean addWord(Word newWord);

	public boolean removeWord(Word newWord);

	public Word getWord(int wordNum);

	public int getSize();

	public boolean searchWordIsDup(String wordId);

	public int searchWordEnglish(String wordInEnglish);

	public int searchWordLang(String wordInLang);

	public int searchWordId(String wordId);

	public List<String> getEngList();

	public boolean isEmpty();

	public boolean isFull();

}

File: GamePortal_PHP_JS/WebContent/WordExplorer/script/WordListInterface.js
/*	ICS499
 *  Group 2
 */

function WordListInterface {
	// Interface Methods
	 var addWord = newWord() //function Word ; //boolean
	 var removeWord = newWord() //function Word; //boolean
	 var getWord(wordNum); //Word
	 var getSize(); //int
	 var searchWordIsDup(wordId); //boolean
	 var searchWordEnglish(wordInEnglish); //int
	 var searchWordLang(wordInLang); //int
	 var searchWordId(wordId); //int
	 var getEngList(); //List<String>
	 var isEmpty(); //boolean
	 var isFull(); //boolean
}